src/Raven.Server/Rachis/Leader.cs

```diff
@@ -712,12 +712,21 @@ public void Initialize()
                 return BlittableResultWriter == null ? r : (r.Index, BlittableResultWriter.Result);
             } 
 
-            public void Dispose()
+            public async ValueTask DisposeAsync()
             {
-                Command.Raw?.Dispose();
-                Command.Raw = null;
-                BlittableResultWriter?.Dispose();
-                _ctxReturn?.Dispose();
+                try
+                {
+                    if (Consumed.Raise() == false)
+                        // If the command has already been dequeued, we must allow it to continue to maintain its context validity.  
+                        await Tcs.Task;
```

Comment:
Do we want to wrap this in `try/catch` and ignore the `catch` ?

Reply:
No, we want to get the Task exception

Reply:
We have two options here - getting here with or without an exception.

Without exception - we for sure want to throw the Task exception.

With exception - basically, if we want to align with the "Without exception," we should throw the first exception (not the one of the Task), but that will require going back to the previous implementation and having this code outside of the Dispose function.
But I think throwing the Task exception here is good enough.